### PR TITLE
Fixed progress widget ignores prior (passed) runs issue

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
- Fixed https://github.com/mitodl/micromasters/issues/2271

#### What's this PR do?
- Check past runs of course i.e if we have passed run in past it will count course as passed.

@pdpinch

### CODE REVIEW

File: static/js/components/ProgressWidget_test.js [hunk 1]
```diff
@@ -41,11 +41,21 @@ export const program = {
...
-          "status": STATUS_NOT_PASSED,
+          "status": STATUS_PASSED,
           "id": 2,
           "course_start_date": "2016-08-22T11:48:27Z",
-          "fuzzy_start_date": "Fall 2017",
+          "fuzzy_start_date": "Fall 2016",
           "course_end_date": "2016-09-09T10:20:10Z"
+        },
+        {
+          "position": 2,
```
Comment (singingwolfboy) [inline static/js/components/ProgressWidget_test.js]: You've added a second run of this course, but the first is passing and the second is not. Based on the way the code used to be, this automated test would pass either way.

Can you modify another one of the courses in this program to have multiple runs, and set the status of the first course run to `STATUS_NOT_PASSED`? That would be an appropriate test of this change.

File: static/js/util/util.js [hunk 2]
```diff
@@ -386,7 +386,7 @@ export function programCourseInfo(program: Program): Object {
   if (program.courses) {
     totalCourses = program.courses.length;
     passedCourses = program.courses.filter(
-      course => course.runs.length > 0 && course.runs[0].status === STATUS_PASSED
+      course => _.some(course.runs, ["status", STATUS_PASSED])
```
Comment (singingwolfboy) [inline static/js/util/util.js]: This works, but I had to look up the documentation for [`_.some`](https://lodash.com/docs#some) to understand what's going on. Can you add a comment to explain this?
```js
// check if any course run has a `status` property set to STATUS_PASSED
course => _.some(course.runs, ["status", STATUS_PASSED])  
```

Comment (amir-qayyum-khan) [inline static/js/util/util.js]: it check if in an array any object has status=== STATUS_PASSED then it will return true. No further check.

Comment (singingwolfboy) [inline static/js/util/util.js]: Yes, I understand that. However, it's not very clear just by looking at the code. That's why I'm asking you to add a comment, as I demonstrated in my comment above.

**Comment by amir-qayyum-khan** [comment]
Done @singingwolfboy

**Comment by singingwolfboy** [comment]
Looks good to me! 👍

**Pull request merged**